plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java

```diff
@@ -17,316 +17,306 @@
-      throws MetaException, NoSuchObjectException {
-    try {
-      Partition partition = this.metastore.getPartition(dbName, tableName, part_vals.get(2));
-      return (partition != null);
-    } catch (MetaException | NoSuchObjectException ex) {
-      throw ex;
+
+    @Override
+    public boolean dropPartition(String dbName, String tableName, List<String> part_vals)
+            throws MetaException, NoSuchObjectException, InvalidObjectException, InvalidInputException {
+        throw new MetaException("Drop partition is not currently supported");
     }
-  }
-
-  @Override
-  public boolean dropPartition(String dbName, String tableName, List<String> part_vals)
-      throws MetaException, NoSuchObjectException, InvalidObjectException, InvalidInputException {
-    throw new MetaException("Drop partition is not currently supported");
-  }
-
-  @Override
-  public List<Partition> getPartitions(String dbName, String tableName, int max)
-      throws MetaException, NoSuchObjectException {
-    try {
-      return this.metastore.getPartitions(dbName, tableName, max);
-    } catch (MetaException | NoSuchObjectException ex) {
-      LOG.error("Get partitions failed table {} database {} error: {}", tableName, dbName, ex.getMessage());
-      throw ex;
+
+    @Override
+    public List<Partition> getPartitions(String dbName, String tableName, int max)
+            throws MetaException, NoSuchObjectException {
+        try {
+            return this.groundTable.getPartitions(dbName, tableName, max);
+        } catch (MetaException | NoSuchObjectException ex) {
+            LOG.error("Get partitions failed table {} database {} error: {}", tableName, dbName, ex.getMessage());
+            throw ex;
+        }
     }
-  }
-
-  @Override
-  public void alterTable(String dbName, String tableName, Table newTable)
-      throws InvalidObjectException, MetaException {
-    try {
-      this.metastore.dropTable(dbName, tableName);
-    } catch (MetaException | InvalidObjectException ex) {
-      LOG.error("Unable to drop previous version of table {} in database {}", tableName, dbName);
-      throw ex;
-    } catch (NoSuchObjectException ex) {
-      LOG.error("Unable to drop previous version of table {} in database {}", tableName, dbName);
-      throw new MetaException("Table " + tableName + " not found in database" + dbName);
-    } catch (InvalidInputException ex) {
-      LOG.error("Invalid input to alter table {} in database {}", tableName, dbName);
-      throw new MetaException("Invalid input to alter table " + tableName + " in database {}" + dbName);
+
+    @Override
+    public void alterTable(String dbName, String tableName, Table newTable)
+            throws InvalidObjectException, MetaException {
+        try {
+            this.groundTable.dropTable(dbName, tableName, EntityState.DELETED.name());
+            this.groundTable.createTableNodeVersion(newTable);
```

Comment:
Doesn't dropping create a new `NodeVersion` that says dropped and then create another `Node` with new attributes? Wouldn't it make more sense to just update the same `Node` with a new version with new attributes?